aspnetcore/security/app-secrets.md

```diff
@@ -1,125 +1,250 @@
+
+> [!WARNING]
+> Environment variables are generally stored in plain, unencrypted text. If the machine or process is compromised, environment variables can be accessed by untrusted parties. Additional measures to prevent disclosure of user secrets may be required.
 
 ## Secret Manager
 
-The Secret Manager tool stores sensitive data for development work outside of your project tree. The Secret Manager tool is a project tool that can be used to store secrets for a .NET Core project during development. With the Secret Manager tool, you can associate app secrets with a specific project and share them across multiple projects.
+The Secret Manager tool stores sensitive data during the development of an ASP.NET Core project. In this context, a piece of sensitive data is an app secret. App secrets are stored in a separate location from the project tree. The app secrets are associated with a specific project or shared across several projects. The app secrets aren't checked into source control.
 
->[!WARNING]
+> [!WARNING]
 > The Secret Manager tool doesn't encrypt the stored secrets and shouldn't be treated as a trusted store. It's for development purposes only. The keys and values are stored in a JSON configuration file in the user profile directory.
 
-## Installing the Secret Manager tool
+## How the Secret Manager tool works
 
-# [Visual Studio](#tab/visual-studio/)
+The Secret Manager tool abstracts away the implementation details, such as where and how the values are stored. You can use the tool without knowing these implementation details. The values are stored in a [JSON](https://json.org/) configuration file in a system-protected user profile folder on the local machine:
 
-Right-click the project in Solution Explorer, and select **Edit \<project_name\>.csproj** from the context menu. 
-Add the highlighted line to the *.csproj* file, and save to restore the associated NuGet package:
+* Windows: `%APPDATA%\Microsoft\UserSecrets\<user_secrets_id>\secrets.json`
+* Linux & macOS: `~/.microsoft/usersecrets/<user_secrets_id>/secrets.json`
 
-[!code-xml[](app-secrets/sample/UserSecrets/UserSecrets-before.csproj?highlight=10)]
+In the preceding file paths, replace `<user_secrets_id>` with the `UserSecretsId` value specified in the *.csproj* file.
 
-Right-click the project in Solution Explorer again, and select **Manage User Secrets** from the context menu. This gesture adds a new `UserSecretsId` node within a `PropertyGroup` of the *.csproj* file, as highlighted in the following sample:
+Don't write code that depends on the location or format of data saved with the Secret Manager tool. These implementation details may change. For example, the secret values aren't encrypted, but could be in the future.
 
-[!code-xml[](app-secrets/sample/UserSecrets/UserSecrets-after.csproj?highlight=4)]
+## Install the Secret Manager tool
```

Comment:
This doesn't apply to 2.1 projects. Can we remove this section completely in the 2.1 version of the docs? Even though you mention this, I think the code sample gets more visibility here. Using DotNetCLiToolReference in 2.1 will lead to a warning.

![image](https://user-images.githubusercontent.com/2696087/40140731-81b1b79c-5908-11e8-9558-f69e7ce5f7c9.png)


https://review.docs.microsoft.com/en-us/aspnet/core/security/app-secrets?view=aspnetcore-2.1&branch=pr-en-us-6373&tabs=visual-studio